src/Microsoft.Extensions.Primitives/StringValues.cs

```diff
@@ -18,17 +19,36 @@ public struct StringValues : IList<string>, IReadOnlyList<string>, IEquatable<St
 
         private readonly string _value;
         private readonly string[] _values;
+        private readonly Encoding _encoding;
+        private readonly byte[] _bytes;
+
+        public StringValues(string value, Encoding encoding)
+        {
+            _value = value;
+            _values = null;
+            _encoding = encoding;
+            _bytes = _encoding.GetBytes(value);
+        }
 
         public StringValues(string value)
         {
             _value = value;
             _values = null;
+            _encoding = null;
+            _bytes = null;
         }
 
         public StringValues(string[] values)
         {
             _value = null;
             _values = values;
+            _encoding = null;
+            _bytes = null;
+        }
+
+        public static StringValues CreateConstant(string value, Encoding encoding)
```

Comment:
I don't think it should have the word "Constant" in it, as constants are a special thing already and this can't create them (despite the intent). I'm happy to just have it be `Create` and `TryGetConstant` be renamed to `TryGetBytes`